Modules/_winapi.c

```diff
@@ -744,6 +744,16 @@ getenvironment(PyObject* environment)
                 "environment can only contain strings");
             goto error;
         }
+        if (PyUnicode_FindChar(key, '\0', 0, PyUnicode_GET_LENGTH(key), 1) != -1 ||
+            PyUnicode_FindChar(value, '\0', 0, PyUnicode_GET_LENGTH(value), 1) != -1)
+        {
+            PyErr_SetString(PyExc_ValueError, "embedded null character");
+            goto error;
+        }
+        if (PyUnicode_FindChar(key, '=', 0, PyUnicode_GET_LENGTH(key), 1) != -1) {
```

Comment:
This search should start at index 1. On Windows, the equals character is allowed as the first character of an environment variable name. It's used for hidden environment variables such as per-drive working directories, e.g. `=E:=E:\working\directory`. 

Really, Windows Python has a deficiency in that `os.environ` is built from the CRT's `environ`, which doesn't include these hidden variables. If a script uses `os.environ.copy()` or `nt.environ.copy()` as the basis for the `env` parameter, then it loses the per-drive working directories (except for the process current directory). For example:

    >>> os.chdir(r'E:\working\directory')
    >>> os.chdir(r'C:\Temp')
    >>> args = ['python', '-c', "import os; print(os.path.abspath('E:test'))"]

    >>> r = subprocess.call(args)
    E:\working\directory\test

    >>> environ = nt.environ.copy()
    >>> r = subprocess.call(args, env=environ)
    E:\test

    >>> environ['=E:'] = r'E:\working\directory'
    >>> r = subprocess.call(args, env=environ)
    E:\working\directory\test